frontend/express/public/javascripts/countly/countly.vue.components.js

```diff
@@ -372,4 +373,2886 @@
+                        });
+                    }
+                    self.customActions.forEach(function(customAction) {
+                        $(self.$refs.dtable).find("tbody").on(
+                            customAction.event,
+                            customAction._columnSelector + " " + customAction.selector,
+                            customAction._handlerFn);
+                    });
+                },
+                refresh: function() {
+                    if (this.isLocked) {
+                        // for pending undo operations
+                        return;
+                    }
+                    if (this.isInitialized && !this.pendingInit) {
+                        CountlyHelpers.refreshTable(this.tableInstance, this.rows);
+                    }
+                    else if (!this.isInitialized && !this.pendingInit) {
+                        this.initialize();
+                    }
+                },
+                softAction: function(row, message, callbacks) {
+                    var self = this;
+                    self.nLocks++;
+                    var undoRow = $("<tr><td class='undo-row' colspan='" + self.finalizedNativeColumns.length + "'>" + message + "&nbsp;<a>" + jQuery.i18n.map["common.undo"] + "</a></td></tr>");
+                    var triggeringRow = $(self.tableInstance).find('tbody tr[data-cly-row-id=' + self.keyFn(row) + ']');
+                    triggeringRow.after(undoRow);
+                    triggeringRow.hide();
+                    var commitWrapped = function() {
+                        undoRow.remove();
+                        self.nLocks--;
+                        callbacks.commit();
+                        self.refresh();
+                    };
+                    var commitTimeout = setTimeout(commitWrapped, 2000);
+                    undoRow.find('a').click(function() {
+                        clearTimeout(commitTimeout);
+                        undoRow.remove();
+                        self.nLocks--;
+                        triggeringRow.show();
+                        if (callbacks.undo) {
+                            callbacks.undo();
+                        }
+                    });
+                },
+                optionEvent: function(action) {
+                    var self = this,
+                        focusedRef = this.focusedRow;
+
+                    this.$emit(action.event, focusedRef, function(options) {
+                        if (options.undo) {
+                            self.softAction(focusedRef, options.undo.message, {
+                                commit: function() {
+                                    self.$emit(options.undo.commit, focusedRef);
+                                }
+                            });
+                        }
+                    });
+                }
+            },
+            template: '<div class="cly-vue-datatable-w" ref="wrapper">\
```

Comment:
Codacy found an issue: [Bad escaping of EOL. Use option multistr if needed.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=6339769)